### DISCUSSION TITLE
The firejail URL open game: Level fireurl unlocked

### DISCUSSION THREAD

**Discussion by rusty-snake** [netblue30/firejail#5364] [General]
Level 0: Nothing is working. URLs do not open.
Level 1: Allow firefox to start and IPC to a running instance to open the URL.
Level 2: [firejail-handler-http] (unlocked by @glitsj16, Jul 2020)
Level 3: [xdg-open.c] (unlocked by @rusty-snake, Mar 2021)
Level 4: [fireurl] (unlocked by @rusty-snake, Sep 2022)
Level 5: 〰〰〰〰〰〰 (unlocked by @〰〰, 〰 〰〰)

Relates to:

* #6462

[firejail-handler-http]: https://github.com/glitsj16/firejail-handler-http
[xdg-open.c]: https://gist.github.com/rusty-snake/5104dc53ce3e52eef86cc34d359aa10e
[fireurl]: https://github.com/rusty-snake/fireurl

**Comment by ghost** [comment]
Interesting and much needed topic.

Just so it is clear from the outset, what would be our `preferred goal` here? I mean, are we going for a non-D-Bus solution, a (limited) D-Bus filtering solution or whatever integrates easiest into the current code-base? Looking forward to the discussion!

**Reply by rusty-snake** [reply]
I see two issues with D-Bus.
1. You will need a to use a library, glib for example.
2. Some programs work with `dbus-user none` but not with `filter`.

**Reply by rusty-snake** [reply]
We should also consider use cases like #3785 to be implemented on top of this in the future.

**Reply by ghost** [reply]
I would prefer a non-D-Bus solution if possible. Besides the security win it would keep profile maintenance to a minimum.

**Reply by rusty-snake** [reply]
If we use D-Bus, it have to be transparent to profiles (maybe an opt out).

---

Anyway, I think that an own private socket per sandbox is a way to follow. It would allow to identify the source of an request and therefore it is possible to implement complexer features on top of it that need per sandbox filtering.

**Reply by topimiettinen** [reply]
Some kind of access control and path sanitization would be nice.

**Reply by ghost** [reply]
For reference, this turned up again in #5566. There's also [firefox-socket](https://github.com/karabaja4/firefox-socket-control). I have been using that before I even knew there was [fireurl](https://github.com/rusty-snake/fireurl).

So all in all there are good options out there as to `how` to fix this. IMO the question we need to focus on is `when` are we going to do this in Firejail and who is the 'we'?
